lib/globs.js

```diff
@@ -23,23 +23,27 @@ const defaultIgnoredByWatcherPatterns = [
 
 const buildExtensionPattern = extensions => extensions.length === 1 ? extensions[0] : `{${extensions.join(',')}}`;
 
-function normalizePatterns(patterns) {
+function normalizePattern(pattern) {
 	// Always use `/` in patterns, harmonizing matching across platforms
 	if (process.platform === 'win32') {
-		patterns = patterns.map(pattern => slash(pattern));
+		pattern = slash(pattern);
 	}
 
-	return patterns.map(pattern => {
-		if (pattern.startsWith('./')) {
-			return pattern.slice(2);
-		}
+	if (pattern.startsWith('./')) {
+		return pattern.slice(2);
+	}
 
-		if (pattern.startsWith('!./')) {
-			return `!${pattern.slice(3)}`;
-		}
+	if (pattern.startsWith('!./')) {
+		return `!${pattern.slice(3)}`;
+	}
 
-		return pattern;
-	});
+	return pattern;
+}
+
+exports.normalizePattern = normalizePattern;
+
+function normalizePatterns(patterns) {
+	return patterns.map(pattern => normalizePattern(pattern));
```

Comment:
Just curious: Why not?
```suggestion
	return patterns.map(normalizePattern);
```

Also, is this function worth it now?

I mean `patterns.map(normalizePattern)` vs. `normalizePatterns(patterns)`?

Reply:
It's a defensive pattern, since `Array#map()` passes two additional arguments. If ever we change `normalizePattern` to take a second argument, for a new use case, it's going to receive the array index for this use case.

Reply:
Fair enough